api/src/main/java/edu/cornell/mannlib/vitro/webapp/controller/authenticate/PasswordChangeRequestSpamMitigation.java

```diff
@@ -0,0 +1,75 @@
+package edu.cornell.mannlib.vitro.webapp.controller.authenticate;
+
+import java.time.LocalDateTime;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class PasswordChangeRequestSpamMitigation {
+
+    private static final Map<String, LocalDateTime> requestHistory = new ConcurrentHashMap<>();
+
+    private static final Map<String, Integer> requestFrequency = new ConcurrentHashMap<>();
+
+    private static final long INTERVAL_INCREASE_MINUTES = 10;
+
+
+    /**
+     * Initializes request history and frequency data for a given email address if it does not exist.
+     *
+     * @param emailAddress The email address for which to initialize request history and frequency data.
+     */
+    private static void initializeHistoryRequestDataIfNotExists(String emailAddress) {
+        requestHistory.putIfAbsent(emailAddress, LocalDateTime.now());
+        requestFrequency.putIfAbsent(emailAddress, 0);
+    }
+
+    /**
+     * Determines whether a password reset request is allowed based on spam mitigation criteria.
+     *
+     * @param emailAddress The email address associated with the password reset request.
+     * @return A PasswordChangeRequestSpamMitigationResponse indicating if the request is allowed.
+     */
+    public static PasswordChangeRequestSpamMitigationResponse isPasswordResetRequestable(String emailAddress) {
```

Comment:
Method name can be better. Maybe not start with is?

Reply:
renamed